app/javascript/__tests__/pages/create-account.test.tsx

```diff
@@ -100,6 +100,80 @@ describe("<CreateAccount />", () => {
     expect(screen.getAllByText("Create an account")).not.toBeNull()
   })
 
+  it("correctly calls the redirect function", async () => {
+    const customLocation = {
+      ...window.location,
+      href: "http://dahlia.com",
+      assign: jest.fn(),
+      replace: jest.fn(),
+      reload: jest.fn(),
+      toString: jest.fn(),
+    }
+
+    Object.defineProperty(window, "location", {
+      configurable: true,
+      enumerable: true,
+      writable: true,
+      value: customLocation,
+    })
+
+    // Redefine the href setter to resolve relative URLs
+    Object.defineProperty(window.location, "href", {
+      configurable: true,
+      enumerable: true,
+      set: function (href: string) {
+        const base = "http://dahlia.com"
+        try {
+          const newUrl = new URL(href, base)
+          this._href = newUrl.href
+        } catch {
+          this._href = href
+        }
+      },
+      get: function () {
+        return this._href || "http://dahlia.com"
+      },
+    })
```

Comment:
thought: this is a lot of setup that doesn't get re-used. We should extract it out if we need ot use it for future tests.